### PR TITLE
Factory - no Windowing stuff

### PR DESCRIPTION
**Pull request by kvark**
This is a cut from #668 without the `Window` trait as it seems controversial at the moment.

The GL part: https://github.com/gfx-rs/gfx_device_gl/pull/23
Examples: https://github.com/gfx-rs/gfx_examples/pull/12

Everything is ready to be reviewed/merged

### CODE REVIEW

**Comment by bvssvni** [comment]
Looks good.

File: src/render/ext/device.rs [hunk 1]
```diff
@@ -71,42 +74,30 @@ impl<D: device::Device> Graphics<D> {
...
 /// Backend extension trait for convenience methods
-pub trait DeviceExt<R: device::Resources, C: device::draw::CommandBuffer<R>>:
-    device::Factory<R> + device::Device<Resources = R, CommandBuffer = C>
-{
-    /// Create a new renderer
-    fn create_renderer(&mut self) -> ::Renderer<R, C>;
+pub trait DeviceExt<D: device::Device, F> {
     /// Convert to single-threaded wrapper
-    fn into_graphics(mut self) -> Graphics<Self>;
+    fn into_graphics(mut self) -> Graphics<D, F>;
```
Comment (bjz) [inline src/render/ext/device.rs]: Why not associated types?

Comment (kvark) [inline src/render/ext/device.rs]: Does it matter? I don't know the guidelines w.r.t associated types, but it's not obvious to me that they'd improve this particular code in any way.

**Comment by bvssvni** [comment]
@kvark Can you fix the `Clone` errors?

**Comment by kvark** [comment]
@bvssvni it was all fixed, I just restarted the build

**Comment by bvssvni** [comment]
Nice!

**Comment by kvark** [comment]
@bvssvni thanks!

**Pull request merged**